### PR TITLE
Additional HTTP Headers

### PR DESCRIPTION
**Pull request by dubeejw**
Allows for additional HTTP headers to be configured for the HTTP client.

### CODE REVIEW

**Comment by dubeejw** [comment]
@mdeuser, please you review.

File: whisk/client_test.go [hunk 1]
```diff
@@ -140,3 +140,31 @@ func TestProxyHost(t *testing.T) {
 		assert.Contains(t, err.Error(), proxyhost, "Setting HTTPS_PROXY to '"+proxyhost+"' did not cause the CLI to use that proxy URL.")
 	}
 }
+
+func TestAdditionalHeaders(t *testing.T) {
+	config := GetValidConfigTest()
+	config.AdditionalHeaders = make(map[string]string)
```
Comment (mdeuser) [inline whisk/client_test.go]: Go http.Header type

File: whisk/client_test.go [hunk 2]
```diff
@@ -140,3 +140,31 @@ func TestProxyHost(t *testing.T) {
...
+	assert.NotNil(t, client)
+
+	newRequest, newRequestErr := client.NewRequest("GET", config.BaseURL.String(), nil, false)
+	assert.Nil(t, newRequestErr, "NewRequest for proxy test failed.")
+	if newRequestErr != nil {
+		fmt.Printf("NewRequest() error: %s\n", newRequestErr.Error())
+	}
+
+	assert.Equal(t, []string([]string{"Value1"}), newRequest.Header["Key1"])
+	assert.Equal(t, []string([]string{"Value2"}), newRequest.Header["Key2"])
```
Comment (mdeuser) [inline whisk/client_test.go]: use Http instance in asserts?  here and below (lins 168,169)

**Comment by dubeejw** [comment]
@mdeuser

**Comment by mdeuser** [comment]
@dubeejw is the intent to provide per-request specific headers? if so, it might be more flexible to include the additional headers via the requestUrl() functions as opposed to the global client configuration

**Comment by dubeejw** [comment]
@mdeuser, AdditionalHeader is meant for consumers of client-go that do not have access to `requestUrl()`.

**Pull request merged**